src/osp/coordinates.h

```diff
@@ -136,21 +172,170 @@ struct CoordinateSpace
                     m_components.at(ccomp_id<CCOMP_T>).value());
     }
 
+    /**
+     * @brief Access multiple components using a TupleCComp
+     *
+     * TupleCComp can be used to conveniently pass CComps as arguments
+     *
+     * @tparam COMP_T... Coordinate components to view
+     */
+    template<typename ... CCOMP_T>
+    constexpr std::optional< TupleCComp<CCOMP_T ...> const>ccomp_view_tuple() const noexcept
+    {
+        // Make sure all CComp IDs are valid indices to m_components
+        if (m_components.size() <= std::max({ccomp_id<CCOMP_T> ...}) )
+        {
+            return std::nullopt;
+        }
+
+        // Make sure all CComps are valid
+        if ( ! (m_components.at(ccomp_id<CCOMP_T>).has_value() && ...))
+        {
+            return std::nullopt;
+        }
+
+        return TupleCComp<CCOMP_T ...>(
+                    Corrade::Containers::arrayCast<typename CCOMP_T::datatype_t>(
+                                        m_components.at(ccomp_id<CCOMP_T>).value()) ...);
+    }
+
     // Queues for systems to communicate
     std::vector<SatToAdd_t> m_toAdd;
-    std::vector<Satellite> m_toRemove;
-    std::vector<Command_t> m_commands;
+    std::vector<uint32_t> m_toRemove;
+    std::vector<Command> m_commands;
 
     // Data and component views are managed by the external system
     // m_data is usually something like CoordspaceCartesianSimple
     entt::any m_data;
     std::vector<std::optional<CoordinateView_t> > m_components;
 
-    Satellite m_center;
-    int16_t m_pow2scale;
+    Satellite m_parentSat;
 
-}; // CoordinateSpace
+    int16_t m_depth{0};
+    int16_t m_pow2scale{10};
+
+}; // struct CoordinateSpace
+
+
+template <typename T>
+constexpr T mulpow2(T in, int16_t pow2) noexcept
+{
+    return (pow2 > 0) ? (in * (1L << pow2)) : (in / (1L << (-pow2)));
```

Comment:
Write comment about how the math here works. And break down the three cases. Less than 0. 0, and greater than 0